model/DeliveryContainerService.php

```diff
@@ -88,27 +89,18 @@ function ($provider) {
      * @param DeliveryExecution $deliveryExecution
      * @return TestPlugin[] the list of plugins
      */
+    /**
+     * @param DeliveryExecution $deliveryExecution
+     * @return array
+     * @throws \common_exception_NotFound
+     */
     public function getPlugins(DeliveryExecution $deliveryExecution)
     {
-        $serviceManager = $this->getServiceManager();
-
-        $pluginService = $serviceManager->get(TestPluginService::SERVICE_ID);
-        $testRunnerFeatureService = $serviceManager->get(TestRunnerFeatureService::SERVICE_ID);
-
+        $pluginService = $this->getServiceLocator()->get(TestPluginService::SERVICE_ID);
         $allPlugins = $pluginService->getAllPlugins();
 
-        $allTestRunnerFeatures = $testRunnerFeatureService->getAll();
-        $activeTestRunnerFeaturesIds = $this->getActiveFeatures($deliveryExecution->getDelivery());
-
-        // If test runner features are defined, we check if we need to disable some plugins accordingly
-        if (count($allTestRunnerFeatures) > 0) {
-            $pluginsToDisable = [];
-            foreach ($allTestRunnerFeatures as $feature) {
-                if (!in_array($feature->getId(), $activeTestRunnerFeaturesIds)) {
-                    $pluginsToDisable = array_merge($pluginsToDisable, $feature->getPluginsIds());
-                }
-            }
-
+        $pluginsToDisable = $this->getPluginsDisabledForDelivery($deliveryExecution->getDelivery());
+        if (count($pluginsToDisable) > 0) {
```

Comment:
Please, use `!empty()` instead of `count()` if you need to determine whether the array is empty or not.

Reply:
Fixed